sway-lib-std/src/chain/auth.sw

```diff
@@ -61,28 +61,28 @@ fn coins_owner() -> Result<Identity, AuthError> {
             i += 1;
         } else {
             // type == InputCoin
-            let input_owner = Option::Some(tx_input_coin_owner(i));
+            let input_owner = tx_input_owner(i);
             if candidate.is_none() {
```

Comment:
Not to be done in this PR necessarily, but I feel a lot of this if/else nesting can be cleaned up a bit using `match`.

Reply:
I'll open an issue.

Reply:
#2237